### PR TITLE
Use pooled array for fp in Xor filter

### PR DESCRIPTION
**Pull request by shubham-sonthalia**

### CODE REVIEW

File: src/Paprika/Utils/Xor8.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public Xor8(IReadOnlyCollection<ulong> keys)
 
         _seed = seed;
 
-        var fp = new byte[arrayLength];
+        IntPtr fp = Marshal.AllocHGlobal(arrayLength);
```
Comment (benaadams) [inline src/Paprika/Utils/Xor8.cs:150]: Better to use `ArrayPool`? Here the allocations are still happening just unmanaged which is a harder heap to clean from fragmentation (making allocations more expesnive and memory use increase over time)

Comment (shubham-sonthalia) [reply src/Paprika/Utils/Xor8.cs:150]: Yeah, makes sense. ArrayPool will give contiguous memory allocation, and that in turn will be more efficient as compared to unmanaged buffer. Thanks for pointing that out.

**Comment by shubham-sonthalia** [comment]
Pls check now.

**Review by benaadams — APPROVED** [review]

**Review by Scooletz — CHANGES_REQUESTED** [review]
This removes one allocation leaving `_fingerprints` in the managed world, but it's always one alloc less :-) 
Please run `dotnet format whitespace` on solution, otherwise builds will break.

**Comment by shubham-sonthalia** [comment]
Done @Scooletz

**Review by Scooletz — APPROVED** [review]
🌶️

**Pull request merged**